internal/controller/apiextensions/composite/environment/selector.go

```diff
@@ -68,28 +68,45 @@ type APIEnvironmentSelector struct {
 	kube client.Client
 }
 
+func isNoOp(ec *v1.EnvironmentConfiguration, currentRefs []corev1.ObjectReference) bool {
+
+	if ec == nil || len(ec.EnvironmentConfigs) == 0 {
+		return true
+	}
+
+	if len(currentRefs) == 0 {
+		return false
+	}
+
+	// currentRefs is non-empty list -> trigger update only if policy.resolve is Always
+	return !ec.Policy.IsResolvePolicyAlways()
+}
+
 // SelectEnvironment for cr using the configuration defined in comp.
 // The computed list of EnvironmentConfig references will be stored in cr.
-func (s *APIEnvironmentSelector) SelectEnvironment(ctx context.Context, cr resource.Composite, rev *v1.CompositionRevision) error {
-	// noop if EnvironmentConfig references are already computed
-	if len(cr.GetEnvironmentConfigReferences()) > 0 {
-		return nil
-	}
-	if rev.Spec.Environment == nil || rev.Spec.Environment.EnvironmentConfigs == nil {
+func (s *APIEnvironmentSelector) SelectEnvironment(ctx context.Context, cr resource.Composite, comp *v1.CompositionRevision) error {
```

Comment:
```suggestion
func (s *APIEnvironmentSelector) SelectEnvironment(ctx context.Context, cr resource.Composite, rev *v1.CompositionRevision) error {
```

Is there a reason to change this variable name? I believe we call revisions `rev` elsewhere.

Reply:
renamed back, this was probably leftover from resolving merge conflicts